### PR TITLE
Support self-vetting of self-asserted tokens

### PR DESCRIPTION
**Pull request by MKodde**
When an Identity is in possession of a vetted self-asserted token. It should be allowed to self-vet successive tokens using that SAT. But only when all tokens are of the SAT type. Once an Identity is in possession of an idenity vetted token (vetted on-premise), all following self vetting tokens must be done with the identity vetted token.

For details see: https://www.pivotaltracker.com/story/show/184292087
And: https://github.com/OpenConext/Stepup-SelfService/pull/284

### CODE REVIEW

File: src/Surfnet/StepupMiddleware/ApiBundle/Authorization/Service/AuthorizationService.php [hunk 1]
```diff
@@ -110,6 +121,51 @@ public function assertRegistrationOfSelfAssertedTokensIsAllowed(IdentityId $iden
...
+        $institution = new Institution((string)$identity->institution);
+        $institutionConfiguration = $this->institutionConfigurationService
+            ->findInstitutionConfigurationOptionsFor($institution);
+        if (!$institutionConfiguration) {
+            return $this->deny('Institution configuration could not be found, unable to ascertain if self-asserted tokens feature is enabled');
+        }
+
+        if (!$institutionConfiguration->selfAssertedTokensOption->isEnabled()) {
+            return $this->deny(sprintf('Institution "%s", does not allow self-asserted tokens', (string) $identity->institution));
+        }
```
Comment (Shaky212) [inline src/Surfnet/StepupMiddleware/ApiBundle/Authorization/Service/AuthorizationService.php:147]: Seems like a repetition of code (r171,r185). Perhaps use a separate method to fetch the institution configuration by the identity?

File: src/Surfnet/StepupMiddleware/CommandHandlingBundle/Identity/CommandHandler/IdentityCommandHandler.php [hunk 2]
```diff
@@ -76,6 +77,7 @@
 use Surfnet\StepupMiddleware\CommandHandlingBundle\Identity\Command\VetSecondFactorCommand;
 use Surfnet\StepupMiddleware\CommandHandlingBundle\Identity\CommandHandler\Exception\DuplicateIdentityException;
 use Surfnet\StepupMiddleware\CommandHandlingBundle\Identity\Service\RegistrationMailService;
+use function var_export;
```
Comment (Shaky212) [inline src/Surfnet/StepupMiddleware/CommandHandlingBundle/Identity/CommandHandler/IdentityCommandHandler.php:80]: Dev left-over?

**Review by Shaky212 — APPROVED** [review]
The described functionality made me a bit dizzy, but by reading it over again and by inspecting the code things found there place and become more clear. Found a few little things and suggestions.

File: src/Surfnet/StepupMiddleware/CommandHandlingBundle/Identity/CommandHandler/IdentityCommandHandler.php (hunk 2 above)
Comment (tvdijen) [reply src/Surfnet/StepupMiddleware/CommandHandlingBundle/Identity/CommandHandler/IdentityCommandHandler.php]: `Use function` actually serves a purpose in terms of performance optimization;
https://veewee.github.io/blog/optimizing-php-performance-by-fq-function-calls/

**Pull request merged**